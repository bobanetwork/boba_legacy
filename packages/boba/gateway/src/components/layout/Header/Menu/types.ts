export interface IMenuItem {
  label: string
  path: string
}

export interface MenuProps {
  isOpen?: boolean
}
