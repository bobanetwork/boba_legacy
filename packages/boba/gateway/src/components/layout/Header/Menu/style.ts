import { Typography } from 'components/global'
import { NavLink } from 'react-router-dom'
import styled from 'styled-components'

export const StyledNav = styled.div`
  display: flex;
  gap: 24px;
`

export const NavLinkItem = styled(NavLink)`
  font-size: 14px;
  line-height: 16px;
  font-weight: 500;
  text-transform: uppercase;
  cursor: pointer;
  text-decoration: none;
  color: ${({ theme }) =>
    theme.name === 'light'
      ? theme.colors['gray'][600]
      : theme.colors['gray'][100]};
  &:hover,
  &.active {
    color: ${({ theme }) =>
      theme.name === 'light'
        ? theme.colors['gray'][800]
        : theme.colors['green'][300]};
  }
`
