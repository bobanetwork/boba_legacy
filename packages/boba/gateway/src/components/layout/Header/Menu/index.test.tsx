import { render } from '@testing-library/react'
import React from 'react'
import CustomThemeProvider from 'themes'
import { Provider } from 'react-redux'
import createMockStore from 'redux-mock-store'
import { MemoryRouter } from 'react-router-dom'
import Menu from '.'

const mockStore = createMockStore()

const renderHeaderMenu = () => {
  return render(
    <MemoryRouter>
      <Provider
        store={mockStore({
          ui: {
            theme: 'dark',
          },
        })}
      >
        <CustomThemeProvider>
          <Menu />
        </CustomThemeProvider>
      </Provider>
    </MemoryRouter>
  )
}

describe('Layout => Header => Menu', () => {
  test('should match the snapshot', () => {
    const { asFragment } = renderHeaderMenu()

    expect(asFragment()).toMatchSnapshot()
  })
})
