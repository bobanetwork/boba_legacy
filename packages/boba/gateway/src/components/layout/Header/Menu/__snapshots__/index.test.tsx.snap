// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout => Header => Menu should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 24px;
}

.c1 {
  font-size: 14px;
  line-height: 16px;
  font-weight: 500;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #A8A8A8;
}

.c1:hover,
.c1.active {
  color: #AEDB01;
}

<div
    class="c0"
  >
    <a
      activeclassname="active"
      class="c1"
      href="/bridge"
    >
      Bridge
    </a>
    <a
      activeclassname="active"
      class="c1"
      href="/history"
    >
      History
    </a>
    <a
      activeclassname="active"
      class="c1"
      href="/earn"
    >
      Earn
    </a>
    <a
      activeclassname="active"
      class="c1"
      href="/stake"
    >
      Stake
    </a>
    <a
      activeclassname="active"
      class="c1"
      href="/DAO"
    >
      Dao
    </a>
  </div>
</DocumentFragment>
`;
